modules/fvad/fvad.c

```diff
@@ -184,26 +198,32 @@ static int decode_update(struct aufilt_dec_st **stp, void **ctx,
 }
 
 
-static bool auframe_vad(Fvad *fvad, struct auframe *af)
+static bool auframe_vad(Fvad *fvad, struct auframe *af, int16_t **buffer)
 {
 	static int chunk_times_ms[] = { 30, 20, 10 };
 
 	int16_t *buf = NULL;
-	int16_t *allocated = NULL;
-	bool detected = false;
 
 	/* convert to 16 bit linear */
 	if (af->fmt == AUFMT_S16LE) {
 		buf = af->sampv;
 	}
 	else {
-		allocated = (int16_t*)mem_alloc(sizeof(int16_t) * af->sampc,
+
+		if (!*buffer) {
+			*buffer = (int16_t*)mem_alloc(sizeof(int16_t) * af->sampc,
 			NULL);
-		buf = allocated;
+
+			if (!*buffer) {
+				warning("fvad: cannot allocate buffer\n");
+				return false;
+			}
```

Comment:
Sorry, I just realized that in case of unsupported sample format module `auconv` should be used. It converts to s16le. So you can remove the else path. Just print a warning in `..._update()` functions and return EINVAL!

There is a config option to bind to s16le for the audio devices. `ausrc_format` and `auplay_format`.